### ISSUE TITLE
Mobile version layout bugs

### ISSUE THREAD

**Issue by l0uden** [mckinsey/vizro#541]
### Description

Here's some configurations where layout is not working as expected:

1. Table in one container, graph in second
<img width="294" alt="image" src="https://github.com/mckinsey/vizro/assets/35569332/f4a6c52f-72d0-4392-b678-340486a39cf5">

2. Table in one container, graph in second in horizontal orientation
<img width="843" alt="image" src="https://github.com/mckinsey/vizro/assets/35569332/ecde59a2-1c55-42bf-a40f-0ef1d117a9fa">

3. Two graphs in horizontal orientation plus card
<img width="841" alt="image" src="https://github.com/mckinsey/vizro/assets/35569332/1e165435-3a88-4714-9659-f873d7d634be">

4. Agrid title overlaps second container tab
<img width="295" alt="image" src="https://github.com/mckinsey/vizro/assets/35569332/f694f78f-88a2-42f6-ac78-d738a0facb1e">

5. Agreed is unreachable with another graph in horizontal orientation 
<img width="840" alt="image" src="https://github.com/mckinsey/vizro/assets/35569332/f8f94bf8-8d96-4d9c-90ab-978aa6f8df0d">

6. No graph displayed with two table on the same page
<img width="290" alt="image" src="https://github.com/mckinsey/vizro/assets/35569332/6a03b35f-171e-4597-970b-fec933359b1c">

7. No graph displayed with lots of components in one container
<img width="292" alt="image" src="https://github.com/mckinsey/vizro/assets/35569332/8d41f06a-ca49-453c-ae9e-41250070ac5a">

### Which package?

vizro

### Package version

0.1.17

### Python version

3.9

### OS

Mac, Linux

### How to Reproduce

Just run tests examples from `vizro-qa`

### Code of Conduct

- [X] I agree to follow the [Code of Conduct](https://github.com/mckinsey/vizro/blob/main/CODE_OF_CONDUCT.md).

**Comment by vladimir-mck** [comment]
**Page 2 – Layout Behavior Explanation**

There are no bugs present on this page.

The layout behavior observed in landscape mode is expected and stems from the way CSS Grid distributes space among its children. The grid layout is defined using a configuration such as `layout=vm.Grid(grid=[[0], [2], [1]])`, which internally maps components to specific grid areas (e.g., `style="grid-area: 3 / 1 / 4 / 2;"`).

However, when the viewport width drops below 768px, the layout switches from CSS Grid to Flexbox. At that point, the explicit grid-area assignments no longer apply, since `grid-area` has no effect in a flex container. As a result, the visual ordering and alignment of elements may differ from their grid layout counterpart.

This is working as intended, and the behavior is dictated by the responsive design logic implemented via CSS media queries.

**Page 3**

Plotly throws an error saying `Something went wrong with axis scaling` when rendering the scatter plot, which appears to be caused by insufficient space for the plot to display properly. I also came across an open [issue](https://github.com/plotly/plotly.js/issues/4155) related to this behavior. Other chart types render without any problems.

<img width="1297" height="504" alt="Image" src="https://github.com/user-attachments/assets/da8423f4-6100-47b4-850c-ef656a1a5dfe" />

**Page 4**

I could not reproduce this issue. See screenshot:

<img width="371" height="348" alt="Image" src="https://github.com/user-attachments/assets/a9fe2094-47f4-49c7-97d5-af3463d7e09b" />

**Comment by petar-qb** [comment]
Should we close the ticket then?

**Comment by l0uden** [comment]
> Should we close the ticket then?

No. There's still Page 1 without investigation and Page 3 is still under investigation.
Also I'll try to reproduce the Page 4 problem.

**Comment by l0uden** [comment]
@vladimir-mck, I've managed to reproduce Page 4 by rotating screen horizontally.

<img width="584" height="430" alt="Image" src="https://github.com/user-attachments/assets/bcfee2e0-8d27-4cd4-a6b4-078e73d295b1" />

**Comment by vladimir-mck** [comment]
@l0uden thanks—I was able to reproduce the issue. It's caused by [this media query](https://github.com/mckinsey/vizro/blob/d5528e847c39a2903d0609b1b6124eab33557380/vizro-core/src/vizro/static/css/layout.css#L204) that reduces the CSS zoom level, which in turn shifts the absolute-positioned elements since their top/left coordinates are scaled independently of the parent container.

To fix this, we should either remove or adjust the zoom override in the media query, or switch to responsive CSS techniques (like adjusting container width, font-size) that won’t affect positioning.
